src/main/python/tests/examples/tutorials/test_adult.py

```diff
@@ -0,0 +1,243 @@
+
+        # Train data
+        X = self.sds.from_numpy( train_data[:train_count])
+        Y = self.sds.from_numpy( train_labels[:train_count])
+
+        # Test data
+        Xt = self.sds.from_numpy(test_data[:test_count])
+        Yt = self.sds.from_numpy(test_labels[:test_count])
+
+        FFN_package = self.sds.source(self.neural_net_src_path, "fnn", print_imported_methods=True)
+
+        network = FFN_package.train(X, Y, 1, 16, 0.01, 1)
+
+        self.assertTrue(type(network) is not None) # sourcing and training seems to works
+
+        FFN_package.save_model(network, '"model/python_FFN/"').compute(verbose=True)
+
+        # TODO This does not work yet, not sure what the problem is
+        #probs = FFN_package.predict(Xt, network).compute(True)
+        # FFN_package.eval(Yt, Yt).compute()
+
+
+    def test_parse_dataset_with_systemdsread(self):
+        # Reduced because we want the tests to finish a bit faster.
+        train_count = 30000
+        test_count = 10000
+        #self.sds.read(self.dataset_path_train, schema=self.dataset_path_train_mtd).compute(verbose=True)
+        print("")
+
+        SCHEMA = '"DOUBLE,STRING,DOUBLE,STRING,DOUBLE,STRING,STRING,STRING,STRING,STRING,DOUBLE,DOUBLE,DOUBLE,STRING,STRING"'
+        F1 = self.sds.read(
+            self.dataset_path_train,
+            schema=SCHEMA
+        )
+        F2 = self.sds.read(
+            self.dataset_path_test,
+            schema=SCHEMA
+        )
+        jspec = self.sds.read(self.dataset_jspec, data_type="scalar", value_type="string")
+        #scaling does not have effect yet. We need to replace labels in test set with the same string as in train dataset
+        X1, M1 = F1.rbind(F2).transform_encode(spec=jspec).compute()
+        col_length = len(X1[0])
+        X = X1[0:train_count, 0:col_length-1]
+        Y = X1[0:train_count, col_length-1:col_length].flatten()-1
+        # Test data
+        Xt = X1[train_count:train_count+test_count, 0:col_length-1]
+        Yt = X1[train_count:train_count+test_count, col_length-1:col_length].flatten()-1
+
+        _ , mean , sigma = scale(self.sds.from_numpy(X), True, True).compute()
+
+        mean_copy = np.array(mean)
+        sigma_copy = np.array(sigma)
+
+        numerical_cols = []
+        for count, col in enumerate(np.transpose(X)):
+            for entry in col:
+                if entry > 1 or entry < 0 or entry > 0 and entry < 1:
+                    numerical_cols.append(count)
+                    break
+
+        for x in range(0,105):
+            if not x in numerical_cols:
+                mean_copy[0][x] = 0
+                sigma_copy[0][x] = 1
+
+        mean = self.sds.from_numpy(mean_copy)
+        sigma = self.sds.from_numpy(sigma_copy)
+        X = self.sds.from_numpy(X)
+        Xt = self.sds.from_numpy(Xt)
+        X = scaleApply(winsorize(X, True), mean, sigma)
```

Comment:
second scaling not needed. if you use the output from the first.

Reply:
also, maybe apply winsorize before getting the scaling? (i'm not sure what the best practice here is)